tokio/src/net/udp/socket.rs

```diff
@@ -1,18 +1,290 @@
 use crate::future::poll_fn;
-use crate::io::PollEvented;
 use crate::net::udp::split::{split, RecvHalf, SendHalf};
 use crate::net::ToSocketAddrs;
 
-use std::convert::TryFrom;
 use std::fmt;
 use std::io;
-use std::net::{self, Ipv4Addr, Ipv6Addr, SocketAddr};
+use std::net::{Ipv4Addr, Ipv6Addr, SocketAddr};
 use std::task::{Context, Poll};
 
 cfg_udp! {
-    /// A UDP socket
-    pub struct UdpSocket {
-        io: PollEvented<mio::net::UdpSocket>,
+    cfg_syscall! {
+        use crate::{syscall::UdpResource, runtime::context};
+        use crate::syscall::Syscalls;
+        use std::sync::Arc;
+
+        fn syscalls() -> Arc<Box<dyn Syscalls>> {
```

Comment:
What is the `Arc<Box<_>>` for? Is it because of Loom? If so, you probably could just use `std::sync::Arc`.

Reply:
This was just because I wasn't sure how to go from `Box` -> `Arc`, but I realize now that there is a `From<Box>` impl for `Arc`. I've updated the PR to use this instead. Thanks!